src/ZPin.cpp

```diff
@@ -515,10 +521,20 @@ void ZPin::pulseWidthEvent(int eventValue)
     this->evCfg->prevPulse = now;
 }
 
-#if 0
-void ZPin::eventCallback()
+void ZPin::pinEventDetected()
 {
-    bool isRise = HAL_GPIO_ReadPin(GPIO_PORT(), GPIO_PIN());
+    bool isRise = gpio_get_pin_level(this->name);
```

Comment:
@mmoskal I couldn't see a latch register for pin interrupts, so this is the best I could do (I think).

We could maybe use EVSYS instead, but that seems to route between peripherals rather than generate useful IRQs.

Reply:
there's something called `PINSTATE`, not sure if that's it though (23.8.13)

Reply:
`PINSTATE` is only useful if debouncing is enabled (tested it out)